chia/consensus/block_header_validation.py

```diff
@@ -43,12 +43,10 @@ def validate_unfinished_header_block(
     blocks: BlockRecordsProtocol,
     header_block: UnfinishedHeaderBlock,
     check_filter: bool,
-    expected_difficulty: uint64,
```

Comment:
this is a little confusing but the naming is important here, its actually not the current difficulty and sub slot iters but the difficulty and sub slot iters expected to be in the block, idk if maybe we should rename ValidationState params to difficulty and sub_slot_iters and add a comment